modules/build/src/main/scala/scala/build/internal/ClassCodeWrapper.scala

```diff
@@ -0,0 +1,62 @@
+package scala.build.internal
+
+case object ClassCodeWrapper extends CodeWrapper {
+  def mainClassObject(className: Name): Name =
+    Name(className.raw ++ "_sc")
+
+  private val userCodeNestingLevel = 1
+  def apply(
+    code: String,
+    pkgName: Seq[Name],
+    indexedWrapperName: Name,
+    extraCode: String,
+    scriptPath: String
+  ) = {
+    val name             = mainClassObject(indexedWrapperName).backticked
+    val wrapperClassName = indexedWrapperName.backticked + "0"
```

Comment:
`$` would be a better suffix here IMO. Usually, the use of `$` in class names says that the class is compiler-generated, or at least, that it's not for end-users to use directly. Which is what we want here.

Reply:
Let's use `$_` since just `$` breaks `package --doc`